### PR TITLE
[`InstructBlip`] FINAL Fix instructblip test

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Fixes the current failing instructblip slow test

This commit: https://github.com/huggingface/transformers/commit/2230d149f0dc4de05ea7a9a1c5295b6d80e495cd was responsible of the test failure however the commit above sillently fixed an issue.

The PR #25105 introduced the correct way of quantizing composable models (blip2, instructblip) and models on the Hub. Before that commit the lm_head of the language model was converted in 8bit, which in fact is wrong. The lm_head should always stay in fp32 for numerical stability and also for consistency with other causal LM models in the library (we keep all lm_head in fp32). Therefore this PR fixes the expected logits and generations

Tested the fix in the latest docker image `huggingface/transformers-all-latest-gpu` on a 2xNVIDIA T4 and the test pass 

cc @ydshieh @amyeroberts

### CODE REVIEW

**Comment by ydshieh** [comment]
Confirmed it passes when running the single test, but failed when running the 2 integration tests together.

@amyeroberts I think we can merge, and I will try to figure out why this strange situation happens. WDYT?

**Comment by younesbelkada** [comment]
> So https://github.com/huggingface/transformers/pull/25105 fixed (among other things) that lm_head was in 8bit, right?

Yes, specifically for models that has submodules that are causalLM models, mainly instructblip and blip2

**Review by ydshieh — APPROVED** [review]
Good on my side. The strange dependency could be addressed separately, but if @younesbelkada  is keen to figure it out, also works for me.

**Review by amyeroberts — APPROVED** [review]
Thanks for digging into this and the detailed explanation!
